wycheproof/wycheproof-tests/src/lib.rs

```diff
@@ -61,6 +61,48 @@ impl ToTokens for SignatureTestVector {
         code.to_tokens(tokens);
     }
 }
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all="camelCase")]
+pub struct XdhTestVector {
+    pub tc_id: u32,
+    pub comment: String,
+    #[serde(with = "hex_serde")]
+    pub public: Vec<u8>,
+    #[serde(with = "hex_serde")]
+    pub private: Vec<u8>,
+    #[serde(with = "hex_serde")]
+    pub shared: Vec<u8>,
+    pub result: ExpectedResult,
+    pub flags: Vec<String>,
+}
+
+impl ToTokens for XdhTestVector {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let tc_id = &self.tc_id;
```

Comment:
Just an idiomatic remark: You can do destructuring here to avoid repetition I think, i.e. `let Self { comment, public, private, shared, result, flags } = self` or similar. Since `&self` mean `self: &Self`, the variables `comment` etc. are then also references.

Reply:
Thanks for this hint, looks much better!